modules/builtin/pulseaudio.go

```
@@ -0,0 +1,110 @@
+package builtin
+
+import (
+	"log"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+	"github.com/mattboll/pulseaudio"
+
+	"fyne.io/fynedesk"
+	wmtheme "fyne.io/fynedesk/theme"
+)
+
+var pulseaudioMeta = fynedesk.ModuleMetadata{
+	Name:        "Pulseaudio",
+	NewInstance: NewPulseaudio,
+}
+
+// Pulseaudio is a progress bar module to modify screen pulseaudio
+type Pulseaudio struct {
+	bar       *widget.ProgressBar
+	client    *pulseaudio.Client
+	soundIcon *widget.Button
+}
+
+// Destroy destroys the module
+func (b *Pulseaudio) Destroy() {
+	b.client.Close()
+}
+
+func (b *Pulseaudio) value() (float32, bool, error) {
+	volume, err := b.client.Volume()
+	if err != nil {
+		return 0, true, err
+	}
+
+	return volume, false, nil
+}
+
+// OffsetValue actually increase or decrease the screen pulseaudio
+func (b *Pulseaudio) OffsetValue(diff int) {
+	floatVal, _, _ := b.value()
```

Error is silently ignored...